// @aws-sdk/client-auto-scaling/dist-es/commands/BatchDeleteScheduledActionCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchDeleteScheduledActionAnswerFilterSensitiveLog, BatchDeleteScheduledActionTypeFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryBatchDeleteScheduledActionCommand, serializeAws_queryBatchDeleteScheduledActionCommand, } from "../protocols/Aws_query";
export class BatchDeleteScheduledActionCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, BatchDeleteScheduledActionCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "AutoScalingClient";
        const commandName = "BatchDeleteScheduledActionCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: BatchDeleteScheduledActionTypeFilterSensitiveLog,
            outputFilterSensitiveLog: BatchDeleteScheduledActionAnswerFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryBatchDeleteScheduledActionCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryBatchDeleteScheduledActionCommand(output, context);
    }
}
