// @aws-sdk/client-auto-scaling/dist-es/commands/DescribeAutoScalingGroupsCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AutoScalingGroupNamesTypeFilterSensitiveLog, AutoScalingGroupsTypeFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDescribeAutoScalingGroupsCommand, serializeAws_queryDescribeAutoScalingGroupsCommand, } from "../protocols/Aws_query";
export class DescribeAutoScalingGroupsCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, DescribeAutoScalingGroupsCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "AutoScalingClient";
        const commandName = "DescribeAutoScalingGroupsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: AutoScalingGroupNamesTypeFilterSensitiveLog,
            outputFilterSensitiveLog: AutoScalingGroupsTypeFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryDescribeAutoScalingGroupsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryDescribeAutoScalingGroupsCommand(output, context);
    }
}
